### ISSUE TITLE
Relevancy loss in multiple word query

### ISSUE THREAD

**Issue by bidoubiwa** [meilisearch/MeiliSearch#1958]
**Describe the bug**

When making a query with multiple word the relevancy of the hits takes a .. hit (hehe). 

Per [the documentation](https://docs.meilisearch.com/learn/core_concepts/relevancy.html#ranking-rules)

> The words rule works from right to left. Therefore, the order of the query string impacts the order of results.

**To Reproduce**
1. Go to [this link](https://codesandbox.io/s/meilisearch-boardgames-uj2tr?file=/src/App.vue)
2. Search for `colon`
3. add a whitespace
4. Look at the results
5. add `d` 
6. Loss in relevancy 

**Expected behavior**
Same or same-ish hits as with `colon `

**Screenshots**
<img width="732" alt="Screenshot 2021-12-01 at 15 44 36" src="https://user-images.githubusercontent.com/33010418/144261479-9a992f34-14f8-4dea-9ff6-eb6638a49ede.png">

<img width="738" alt="Screenshot 2021-12-01 at 16 21 09" src="https://user-images.githubusercontent.com/33010418/144261854-a840f067-1e1a-411f-96ec-f0901ed80188.png">

<img width="905" alt="Screenshot 2021-12-01 at 16 22 59" src="https://user-images.githubusercontent.com/33010418/144262288-7e3b3d9e-ac36-43fa-929a-a53281f00eeb.png">

**MeiliSearch version:** v0.23.1

**Comment by curquiza** [comment]
Not expected as far as I understand.
We should investigate, fix and add tests regarding this for v0.26.0.

**Comment by bidoubiwa** [comment]
Thanks to @Kerollmops he found out it was because of `boardgameartist` that I did not check: 
<img width="487" alt="Screenshot 2021-12-13 at 13 13 29" src="https://user-images.githubusercontent.com/33010418/145810833-f6cf1ea3-f4f2-4e02-823e-059a26e895c2.png">

To avoid that I should change the ranking rules the following way: 
```
  "rankingRules": [
    "words",
    "typo",
    "attribute", // attribute before proximity
    "proximity",
    "sort",
    "exactness",
    "numberRatings:desc",
    "yearPublished:desc"
  ],
```
